### PR TITLE
Feature/multitopic

### PR DESCRIPTION
**Pull request by wnichols**
Modified the subscriber to allow it to subscribe to multiple topics on the same cluster.  Also moved parameters specific to the cluster connection from the "handlers" section to the "settings" section of the config.  This was the consensus of the mashling group.  This is preparatory to Anant merging his conditional switching logic into the subscriber.

@mellis
@fm-tibco

### CODE REVIEW

File: trigger/kafkasub/trigger.go [hunk 1]
```diff
@@ -119,49 +121,53 @@ func run(t *KafkaSubTrigger) {
...
-	if t.kafkaParms.partitions == nil { //subscribe to all valid partitions
-		t.kafkaParms.partitions = validPartitions
-		for _, part := range validPartitions {
-			startConsumer(t, part)
+	consumers := make(map[string]sarama.PartitionConsumer)
+	t.partitionConsumers = &consumers
+	for id, handler := range t.kafkaParms.handlers {
+		validPartitions, error := kafkaConsumer.Partitions(handler.topic)
+		if error != nil {
+			panic(fmt.Errorf("Failed to get valid partitions for topic [%s] for reason [%s].  Aborting subscriber",
```
Comment (torresashjian) [inline trigger/kafkasub/trigger.go]: Start() in a trigger should not panic, just return an error and the engine will handle it.

File: trigger/kafkasub/trigger.go [hunk 2]
```diff
@@ -119,49 +121,53 @@ func run(t *KafkaSubTrigger) {
...
 			}
-			flogoLogger.Errorf("Configured partition [%d] on topic [%s] does not exist", confPart, t.kafkaParms.topic)
 		}
-	}
-	if len(*t.partitionConsumers) < 1 {
-		panic(fmt.Errorf("Kafka consumer is not configured for any valid partitions"))
-	} else {
-		flogoLogger.Debugf("Kafka consumers for topic [%s] started", t.kafkaParms.topic)
+		if len(*t.partitionConsumers) < 1 {
+			panic(fmt.Errorf("Kafka consumer is not configured for any valid partitions"))
```
Comment (torresashjian) [inline trigger/kafkasub/trigger.go]: Start() in a trigger should not panic, just return an error and the engine will handle it.

File: trigger/kafkasub/trigger.go [hunk 3]
```diff
@@ -119,49 +121,53 @@ func run(t *KafkaSubTrigger) {
...
+		} else {
```
Comment (torresashjian) [inline trigger/kafkasub/trigger.go]: Is this else needed?

File: trigger/kafkasub/trigger.go [hunk 4]
```diff
@@ -210,111 +216,90 @@ func initKafkaParms(t *KafkaSubTrigger) error {
...
 		t.kafkaParms.brokers[brokerNo] = broker
 	}
+	//clientKeystore
+	/*
+		Its worth mentioning here that when the keystore for kafka is created it must support RSA keys via
+		the -keyalg RSA option.  If not then there will be ZERO overlap in supported cipher suites with java.
+		see:   https://issues.apache.org/jira/browse/KAFKA-3647
+		for more info
+	*/
+	if t.config.Settings["truststore"] != nil {
```
Comment (torresashjian) [inline trigger/kafkasub/trigger.go]: No need to change it, but have a look at "The expression may be preceded by a simple statement" 
in https://golang.org/ref/spec#If_statements, it is a cool feature :)

File: trigger/kafkasub/trigger.go [hunk 5]
```diff
@@ -223,8 +222,7 @@ func initKafkaParms(t *KafkaSubTrigger) error {
 		see:   https://issues.apache.org/jira/browse/KAFKA-3647
 		for more info
 	*/
-	if t.config.Settings["truststore"] != nil {
-		trustStore := t.config.Settings["truststore"]
+	if trustStore := t.config.Settings["truststore"]; trustStore != nil {
 		if trustStore != nil && len(trustStore.(string)) > 0 {
```
Comment (torresashjian) [inline trigger/kafkasub/trigger.go]: You are checking trustStore != nil twice

**Comment by wnichols** [comment]
No suggestion too small :)  Though I should have notice that but I'm in a dreadful hurry because of escalations.

**Pull request merged**